sale_order_line_position/models/ir_actions_report.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
+        self.sale_recompute_position(res_ids)
```

Comment:
Is this really ? Launching code at report call ?

Reply:
My question is: Is it programmatically acceptable to do so ?

Reply:
Ah, you mean a record should not risk being changed when printed ?
I can understand your point. But in this situation it is only information that is read-only that will be changed so I would think not that bad a practice.